### ISSUE TITLE
Disabling BatteryMonitoring

### ISSUE THREAD

**Issue by serjooo** [dennisweissmann/DeviceKit#116]
Hello,

First of all thank you very much for this great replacement of UIDevice!

I was looking at the code and noticed something and was puzzled on why while intializing a `BatteryState` enum case you would set `isBatteryMonitoringEnabled` to true then at the end set it back to off. Is there any specific reason for this? Because I had this scenario, where I personally set this to true myself and added a notification observer for `UIDeviceBatteryLevelDidChange`, however, since afterwards I would be calling `Device.batteryLevel` it would go ahead initialize a `BatteryState` and set the `isBatteryMonitoringEnabled` back to false, thus I would never receive a callback from the notification.

Here is the code where this specifically happens line 652 - 663:

```
fileprivate init() {
        UIDevice.current.isBatteryMonitoringEnabled = true
        let batteryLevel = Int(round(UIDevice.current.batteryLevel * 100)) // round() is actually not needed anymore since -[batteryLevel] seems to always return a two-digit precision number
        // but maybe that changes in the future.
        switch UIDevice.current.batteryState {
        case .charging: self = .charging(batteryLevel)
        case .full: self = .full
        case .unplugged:self = .unplugged(batteryLevel)
        case .unknown: self = .full // Should never happen since `batteryMonitoring` is enabled.
        }
        UIDevice.current.isBatteryMonitoringEnabled = false
      }
```

**Comment by dennisweissmann** [comment]
Hey :) thanks for the kind words!

The reason is that I assumed that monitoring is an expensive task (because otherwise there wouldn’t be a need to enable/disable it, right?).

But your use case is definitely valid and I’ll try to see if there’s a Notification Center API I can use, even though that’s a anychrounous task then ...

Thanks for reporting this issue, I’ll try to find a workaround (probably with new API 😥) ASAP.

**Comment by serjooo** [comment]
> The reason is that I assumed that monitoring is an expensive task (because otherwise there wouldn’t be a need to enable/disable it, right?).

I'm not sure either if it is considered an expensive task or not. Apple docs don't say much about this either. However, they do mention that:

> For this notification to be sent, you must set the isBatteryMonitoringEnabled property to true.
> Notifications for battery level change are sent no more frequently than once per minute. Do not attempt to calculate battery drainage rate or battery time remaining; drainage rate can change frequently depending on built-in applications as well as your application.

[Apple Docs](https://developer.apple.com/documentation/foundation/nsnotification.name/1620060-uidevicebatteryleveldidchange)

It would just be okay I guess if you would allow the developer to customize the behavior with a static property to keep the monitoring enabled or disable it after receiving the battery level, as he would be aware of the cost and requires it in his/her project.

**Comment by dennisweissmann** [comment]
Hey @serjooo :) Do you think #118 is a good solution?

**Comment by serjooo** [comment]
@dennisweissmann  Perfect solution! Would work like a charm! However, I think behavior needs documentation. The behavior is not very obvious unless someone reads through the code.

**Comment by dennisweissmann** [comment]
Fixed in #118
